homeassistant/components/mqtt/__init__.py

```diff
@@ -54,6 +57,7 @@
 DEFAULT_QOS = 0
 DEFAULT_RETAIN = False
 DEFAULT_PROTOCOL = PROTOCOL_311
+DEFAULT_TLS_INSECURE = True
```

Comment:
This default is not being used and neither should it ever be defaulting to be insecure. As it's not being used, you can remove it.